### PR TITLE
Quota limiter: make method to delay cpu more precise

### LINKED ISSUE #12218: Quota limiter: correct the mehod to limit cpu
### What version of TiKV are you using?

https://github.com/tikv/tikv/commit/9a43faf4da20389a4e2c262ee8ab8b369a3bfec4. After https://github.com/tikv/tikv/pull/12151

### What operating system and CPU are you using?

don't matter

### Steps to reproduce

don't matter

### What did you expect?
1. The `foreground-cpu-time` configured by the user will be reduced by a fixed factor, which is not necessary.
2. The result returned by cpu_limiter does not need to be processed additionally.

### What did happened?
original implement don't affect the correctness of the results.

### PR DESCRIPTION
**Pull request by hunterlxt**
Signed-off-by: Xintao <hunterlxt@live.com>

### What is changed and how it works?

Issue Number: Close #12218

What's Changed:

```commit-message
Use the results returned by the cpu limiter directly without extra processing
```

### Related changes

- Need to cherry-pick to the release 6.0 branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- No

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hunterlxt** [comment]
In recent internal bench tests, I found that refill duration can affect performance, and less refill duration can get less delay duration when the bucket value is used off, which make grpc p99 lower and qps higher. But less refill duration means that more requests will be delayed (Each delay is shortened), so context switch may increase. 

Here are some results:
pqHYfUHmsl
0XQfCsp8yN
6hPqAgG2Kp
N2MgarRbwK

**Comment by hunterlxt** [comment]
In the lasted commit, I added a new config to allow user to set max delay for each single request, which is a result of discussion with @tonyxuqqi

File: components/tikv_util/src/quota_limiter.rs [hunk 1]
```diff
@@ -95,11 +89,14 @@ impl QuotaLimiter {
         cpu_quota: usize,
         write_bandwidth: ReadableSize,
         read_bandwidth: ReadableSize,
+        max_delay_duration: ReadableDuration,
     ) -> Self {
         let cputime_limiter = if cpu_quota == 0 {
             Limiter::new(f64::INFINITY)
         } else {
-            Limiter::new(cpu_quota as f64 * CPU_TIME_FACTOR * 1000_f64)
+            <Limiter>::builder(cpu_quota as f64 * 1000_f64)
```
Comment (sticnarf) [inline components/tikv_util/src/quota_limiter.rs:97]: The `<>` looks unnecessary?

**Review by sticnarf — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**